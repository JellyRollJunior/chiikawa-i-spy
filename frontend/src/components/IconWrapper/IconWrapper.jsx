import styles from './IconWrapper.module.css';
import chiikawa from '../../assets/characterIcons/chiikawa-icon.png';
import hachiware from '../../assets/characterIcons/hachiware-icon.png';
import kurimanju from '../../assets/characterIcons/kurimanju-icon.png';
import momonga from '../../assets/characterIcons/momonga-icon.png';
import rakko from '../../assets/characterIcons/rakko-icon.png';
import usagi from '../../assets/characterIcons/usagi-icon.png';

const icons = [chiikawa, hachiware, kurimanju, momonga, rakko, usagi];

const IconWrapper = ({ size, children, leftIcon = 0, rightIcon = 1 }) => {
  return (
    <div className={styles.iconWrapper}>
      <img
        src={icons[leftIcon]}
        style={{ height: size, width: size }}
        alt="Chiikawa icon"
      />
      {children}
      <img
        src={icons[rightIcon]}
        style={{ height: size, width: size }}
        alt="Hachiware icon"
      />
    </div>
  );
};

export { IconWrapper };
