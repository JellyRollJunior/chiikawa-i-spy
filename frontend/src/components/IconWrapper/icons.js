import chiikawa from '../../assets/characterIcons/chiikawa-icon.png';
import hachiware from '../../assets/characterIcons/hachiware-icon.png';
import kurimanju from '../../assets/characterIcons/kurimanju-icon.png';
import momonga from '../../assets/characterIcons/momonga-icon.png';
import rakko from '../../assets/characterIcons/rakko-icon.png';
import usagi from '../../assets/characterIcons/usagi-icon.png';

const icons = [chiikawa, hachiware, kurimanju, momonga, rakko, usagi];

export { icons };
