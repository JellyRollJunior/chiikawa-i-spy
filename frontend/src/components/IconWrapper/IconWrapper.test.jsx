import { describe, it, expect } from 'vitest';
import { screen, render } from '@testing-library/react';
import { IconWrapper } from './IconWrapper.jsx';
import { icons } from './icons.js';

describe('IconWrapper component', () => {
  it('renders icons to left and right of child element', () => {
    const { container } = render(
      <IconWrapper>
        <button>I am a child!</button>
      </IconWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders icons based on icon index', () => {
    const leftIndex = 2;
    const rightIndex = 3;
    render(<IconWrapper leftIcon={leftIndex} rightIcon={rightIndex} />);

    const leftIconImg = screen.getAllByRole('img')[0].src.split('/').pop();
    const rightIconImg = screen.getAllByRole('img')[1].src.split('/').pop();
    expect(leftIconImg).toStrictEqual(icons[leftIndex].split('/').pop());
    expect(rightIconImg).toStrictEqual(icons[rightIndex].split('/').pop());
  });
});
