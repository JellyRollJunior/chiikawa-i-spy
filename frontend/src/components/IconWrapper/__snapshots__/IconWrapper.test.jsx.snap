// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconWrapper component > renders icons to left and right of child element 1`] = `
<div>
  <div
    class="_iconWrapper_5d4a96"
  >
    <img
      alt="Chiikawa character icon"
      src="/src/assets/characterIcons/chiikawa-icon.png"
      style="margin-right: 12px;"
    />
    <button>
      I am a child!
    </button>
    <img
      alt="Chiikawa character icon"
      src="/src/assets/characterIcons/hachiware-icon.png"
      style="margin-left: 12px;"
    />
  </div>
</div>
`;
